samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -0,0 +1,251 @@
+      console.log('Original Error', e.error?.details?.sourceError);
+    }
+  }
+
+  /**
+   * Decides whether to display video or play audio based on the track type.
+   * @param {Object} event - The track event object.
+   */
+  displayTrack(event) {
+    if (event.track.kind === 'video') {
+      this.displayVideo(event);
+    } else {
+      this.playAudio(event);
+    }
+  }
+
+  /**
+   * Adds a <video> element to the DOM and attaches the video track to it.
+   * @param {Object} event - The track event object containing the video track.
+   */
+  displayVideo(event) {
+    let videosDiv = document.getElementById('videos');
+    let videoEl = document.createElement('video');
+    videosDiv.appendChild(videoEl);
+    videoEl.style.width = '100%';
+    videoEl.srcObject = new MediaStream([event.track]);
+    videoEl.play();
+  }
+
+  /**
+   * Adds an <audio> element to the DOM and attaches the audio track to it.
+   * @param {Object} event - The track event object containing the audio track.
+   */
+  playAudio(event) {
+    if (event.participant.local) {
+      return; // Avoid playing local audio
+    }
+    let audioEl = document.createElement('audio'); // Create <audio> element
+    document.body.appendChild(audioEl); // Add <audio> element to DOM
+    audioEl.srcObject = new MediaStream([event.track]); // Attach audio track to <audio> element
+    audioEl.play(); // Play audio track
+  }
+
+  /**
+   * Removes the track's element (audio or video) from the DOM when it stops.
+   * @param {Object} event - The track event object.
+   */
+  destroyTrack(event) {
+    let tracks = document.querySelectorAll('video, audio');
+    tracks.forEach((el) => {
+      if (el.srcObject && el.srcObject.getTracks().includes(event.track)) {
+        el.remove();
+      }
+    });
+  }
+
+  /**
+   * Updates UI for track state changes (e.g., mute states).
+   */
+  handleParticipantStateUpdate(event) {
+    this.getTrackStates();
+  }
+
+  /**
+   * Updates the UI with the current active speaker's identity.
+   * @param {Object} event - The active speaker change event object.
+   */
+  handleActiveSpeakerChange(event) {
+    document.getElementById(
+      'active-speaker'
+    ).textContent = `Active Speaker: ${event.activeSpeaker.peerId}`;
+  }
+
+  /**
+   * Tries to join a call with provided room URL and optional join token.
+   * @param {string} roomUrl - The URL of the room to join.
+   * @param {string|null} joinToken - An optional token for joining the room.
+   */
+  async joinRoom(roomUrl, joinToken = null) {
+    if (!roomUrl) {
+      console.error('Room URL is required to join a room.');
+      return;
+    }
+
+    const joinOptions = { url: roomUrl };
+    if (joinToken) {
+      joinOptions.token = joinToken;
+      console.log('Joining with token:', joinToken);
+    } else {
+      console.log('Joining without a token.');
+    }
+
+    try {
+      await this.call.join(joinOptions);
+      console.log(`Successfully joined: ${roomUrl}`);
```

Comment:
Just a thought: I wonder if using the `"joined-meeting"` event instead of printing this after the await might be nicer and more consistent. Then we show _one_ consistent flow for when a participant joins a meeting, instead of splitting it between the event listener _and_ await approaches.

Reply:
Good call 👍